### DISCUSSION TITLE
how to add OCI docker repo into argocd

### DISCUSSION THREAD

**Discussion by reddy2018** [argoproj/argo-cd#12353] [Q&A]
one of my client helm chats in docker hub repo. 
i have created a secret to add the repository and  its failed

here is my yaml file
apiVersion: v1
kind: Secret
metadata:
  name: wrm5
  namespace: argocd
  labels:
    argocd.argoproj.io/secret-type: repository
stringData:
  enableOCI: "true"
  type: helm
  url: https://registry-1.docker.io/enghousenetworksbss

can any one guide me to add this repo

**Comment by reddy2018** [comment]
i have fix it. thanks

**Reply by alexef** [reply]
what was the fix?

**Reply by varun-kobai** [reply]
I had the same issue, I resolved it by adding the `ForceHttpBasicAuth: "true"` key in the `StringData` field. Not sure how this necessarily affects things wrt security best practices, but it works for now.

**Comment by comtihon** [comment]
you can use this terraform resource:
```
resource "kubernetes_secret" "argocd-helm-repo-secret" {
  metadata {
    name      = "argocd-helm-registry"
    namespace = "argocd"
    labels    = {
      "argocd.argoproj.io/secret-type" : "repository"
    }
  }

  data = {
    enableOCI = "true"
    name : "argocd-helm-registry"
    username : "_json_key_base64"
    password : filebase64("<service acc.json>")
    type : "helm"
    url : "europe-west3-docker.pkg.dev/<project name>/<registry name>/${var.environment}"
  }

  type = "Opaque"
}```
